src/app/shared/components/chat-bot/chat-bot.component.ts

```diff
@@ -0,0 +1,291 @@
+import {
+  AfterViewChecked,
+  Component,
+  ElementRef,
+  OnDestroy,
+  OnInit,
+  ViewChild,
+} from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { Observable, Subscription, forkJoin } from 'rxjs';
+import { mergeMap } from 'rxjs/operators';
+import { AppStateInterface } from 'src/app/models/appState.interface';
+import { ChatbotFAQInterface } from 'src/app/models/chatbotFAQ.interface';
+import { MessageInterface } from 'src/app/models/message.interface';
+import { ProductInterface } from 'src/app/models/product.interface';
+import { ChatbotService } from 'src/app/services/chatbot.service';
+import * as CartAction from '../../../store/actions/CartActions';
+import { ProductService } from 'src/app/services/product.service';
+
+enum Tags {
```

Comment:
you can create separate file chat-bot.constants.ts and export them and read from there in order to have component more clear